### PR TITLE
LPS-145163 [CP 2.0] Create Activation Form Card Component

### PR DESCRIPTION
**Pull request by joaosvassis**

### CODE REVIEW

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js [hunk 1]
```diff
@@ -201,8 +192,8 @@ export const getAccountRolesAndAccountFlags = gql`
 `;
 
 export const getAccountRoles = gql`
-	query getAccountRoles($accountId: Long!) {
-		accountAccountRoles(accountId: $accountId) {
+	query getAccountRoles($filter: Long!) {
+		accountAccountRoles(accountId: $filter) {
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js:196]:
```suggestion
	query getAccountRoles($accountId: Long!) {
		accountAccountRoles(accountId: $accountId) {
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+import client from '../../../../apolloClient';
+import {getAccountSubscriptionsTerms} from '../../../../common/services/liferay/graphql/queries';
+import {getCurrentEndDate} from '../../../../common/utils/';
+import {useCustomerPortal} from '../../context';
+import {status} from '../../utils/constants';
+import StatusTag from '../StatusTag';
+
+const ActivationStatus = () => {
+	const [
+		{assetsPath, project, subscriptionGroups, userAccount},
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js:23]:
```suggestion
const ActivationStatus = ({project, subscriptionGroups, userAccount}) => {
	const [
		{assetsPath},
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+			<h2>Activation Status</h2>
+			<p className="font-weight-normal text-neutral-7 text-paragraph">
+				{activationStatus.subtitle}
+			</p>
+			<div>
+				<ClayCard className="activation-status-container m-0 rounded">
+					<ClayCard.Body className="pb-3 pl-4 pr-4 pt-3">
+						<div className="align-items-center d-flex position-relative">
+							<img
+								className={`mr-5 ${activationStatus.className}`}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js:132]: Try to use the className lib

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+								width={34.36}
+							/>
+
+							<ClayCard.Description
+								className="h5"
+								displayType="title"
+								tag="h5"
+								title={project?.name}
+							>
+								{project?.name}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js:145]:
```suggestion
								tag="h5"
								title={project.name}
							>
								{project.name}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+	] = useCustomerPortal();
+	const [activationStatusDate, setActivationStatusDate] = useState('');
+
+	const currentActivationStatus = {
+		'Active': {
+			className: 'img-activation-status-active',
+			link: () => (
+				<a
+					className="font-weight-semi-bold m-0 p-0 text-brand-primary text-paragraph"
+					href={`https://console.liferay.cloud/projects/${project?.code}/overview`}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js:33]:
```suggestion
					href={`https://console.liferay.cloud/projects/${project.code}/overview`}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+					<ClayIcon className="ml-1" symbol="order-arrow-right" />
+				</a>
+			),
+			message:
+				'Your DXP Cloud environments are ready. Go to the Product Console to view DXP Cloud details.',
+			status: status.active,
+		},
+		'In Progress': {
+			className: 'img-activation-status-in-progress',
+			link: () => {},
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js:47]: This param it's necessary?
```suggestion
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+						<a className="btn btn-link font-weight-semi-bold m-0 p-0 text-brand-primary text-paragraph">
+							Finish Activation
+							<ClayIcon
+								className="ml-1"
+								symbol="order-arrow-right"
+							/>
+						</a>
+					)}
+				</>
+			),
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js:66]:
```suggestion
			link: () => userAccount.isAdmin && (
				<a className="btn btn-link font-weight-semi-bold m-0 p-0 text-brand-primary text-paragraph">
							Finish Activation
							<ClayIcon
								className="ml-1"
								symbol="order-arrow-right"
							/>
						</a>
			),
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+				setActivationStatusDate(
+					`${getCurrentEndDate(
+						earliestStartDate
+					)} - ${getCurrentEndDate(farthestEndDate)}`
+				);
+			}
+		};
+
+		getSubscriptionTerms();
+	}, [project.accountKey]);
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js:119]:
```suggestion
	}, [project]);
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+								height={31.98}
+								src={`${assetsPath}/assets/navigation-menu/dxp_icon.svg`}
+								width={34.36}
+							/>
+
+							<ClayCard.Description
+								className="h5"
+								displayType="title"
+								tag="h5"
+								title={project?.name}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js:143]:
```suggestion
								title={project.name}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js (hunk 4 above)
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/index.js:145]:
```suggestion
								{project.name}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/DXPCloud/index.js [hunk 10]
```diff
@@ -8,9 +8,10 @@
  * permissions and limitations under the License, including but not limited to
  * distribution rights of the Software.
  */
+import ActivationStatus from '../../components/ActivationStatus/index';
 
 const DXPCloud = () => {
-	return <div>DXP Cloud page</div>;
+	return <ActivationStatus />;
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/DXPCloud/index.js:14]: Here pass the values with context

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Comment by jairmedeiros** [comment]
Just starting reviewing :)

**Comment by jairmedeiros** [comment]
resent https://github.com/pt-liferay-solutions/liferay-portal/pull/723

**Pull request closed without merging**